lib/DDG/Spice/SkyscannerFlightSearch.pm

```diff
@@ -0,0 +1,113 @@
+package DDG::Spice::SkyscannerFlightSearch;
+
+# ABSTRACT: This IA will call Skyscanner's API to retrieve the cheapest flight prices to 
+# different countries from the user's country
+
+use DDG::Spice;
+use strict;
+use warnings;
+use JSON;
+
+# No caching of results
+spice is_cached => 0;
+
+spice wrap_jsonp_callback => 1; # only enable for non-JSONP APIs (i.e. no &callback= parameter)
+
+spice from => '([^/]*)/([^/]*)/([^/]*)/([^/]*)/([^/]*)';
+spice to => 'http://partners.api.skyscanner.net/apiservices/browseroutes/v1.0/$1/$3/$2/$4/$5/anytime/anytime?apikey={{ENV{DDG_SPICE_SKYSCANNER_APIKEY}}}';
+spice alt_to => {
+    skyscanner_images => {
+        to => 'https://gateway.skyscanner.net/travel-api/v1/entities?external_ids=$1&enhancers=images&apikey={{ENV{DDG_SPICE_SKYSCANNER_IMAGES_APIKEY}}}'
+    }
+};
+
+triggers startend => 'skyscanner';
+
+# Load the list of currencies for each market
+my $currencies_raw = share('currencyCountryMapping.json')->slurp;
+my $currencies = decode_json($currencies_raw);
+
+# Load the list of countries and their ISO code
+my $countries_raw = share('countryIsoCodes.json')->slurp;
+my $countries = decode_json($countries_raw);
+
+# Load the list of cities and their Skyscanner code (using Skyscanner code as IATA code is ambiguous in certain cases such as Berlin where the city 
+# and one of the airports have the same code)
+my $cities_raw = share('citySkyscannerCodes.json')->slurp;
+my $cities = decode_json($cities_raw);
+
+# Handle statement
+handle remainder => sub {
+    my $origin = "";
+    my $destination = "";
+    
```

Comment:
@samheyman I don't know how often the $loc is not available. I should be there for the majority of places.

In cases where we don't have it, we should just ignore the query though. All major cities and countries are supported as far as I know